klarna-on-demand/src/main/java/com/klarna/ondemand/RegistrationActivity.java

```diff
@@ -24,9 +30,19 @@ protected String getUrl() {
 
     @Override
     protected void handleUserReadyEvent(Map<Object, Object> payload) {
-        String token = (String) payload.get("userToken");
         Intent result = new Intent();
-        result.putExtra(EXTRA_USER_TOKEN, token);
+
+        // Backwards compatibility.
+        result.putExtra(EXTRA_USER_TOKEN, (String)payload.get(PAYLOAD_USER_TOKEN));
+
+        Map<?, ?> userDetails =  payload.containsKey(PAYLOAD_USER_DETAILS) ? (Map<?, ?>)payload.get(PAYLOAD_USER_DETAILS) : (Map<?, ?>)Collections.EMPTY_MAP;
```

Comment:
No need for the default value.
Payload is designed to always contain a HASH of 'userDetails'.